crates/bdk/src/wallet/mod.rs

```diff
@@ -428,27 +429,115 @@ impl<D> Wallet<D> {
             .next()
     }
 
-    /// Return a single transactions made and received by the wallet
+    /// Inserts the given foreign `TxOut` at `OutPoint` into the wallet's transaction graph. The
+    /// inserted foreign TxOut is persisted.
     ///
-    /// Optionally fill the [`TransactionDetails::transaction`] field with the raw transaction if
-    /// `include_raw` is `true`.
-    pub fn get_tx(&self, txid: Txid, include_raw: bool) -> Option<TransactionDetails> {
+    /// Only insert TxOuts you trust the values for!
+    pub fn insert_txout(&mut self, outpoint: OutPoint, txout: TxOut) -> Result<(), D::WriteError>
+    where
+        D: PersistBackend<ChangeSet>,
+    {
+        let additions = self.indexed_graph.insert_txout(outpoint, &txout);
+        self.persist.stage(ChangeSet::from(additions));
+        self.persist.commit()?;
```

Comment:
We are not meant to commit here. What if you want to insert a bunch of txouts? This would do IO for every single call.

Reply:
Makes sense. I removed the commit() and Result<>, and updated the docs.